### PR TITLE
refactor `/transactions`, `/rawtx` & `/fee` endpoints

### PR DESCRIPTION
**Pull request by fuxingloh**
#### What kind of PR is this?:

/kind refactor

#### What this PR does / why we need it:

Remapped endpoints:

1. Moved `/blocks/:hash/transactions/:id/vins` to `/transactions/:id/vins` as txid is unique, we don't need to nest it
   under blocks
1. Moved `/blocks/:hash/transactions/:id/vouts` to `/transactions/:id/vouts` as txid is unique, we don't need to nest it
   under blocks
1. Moved `/transactions` to `/rawtx/send` to keep API design symmetric as we are dealing with rawtx not transactions
1. Moved `/transactions/test` to `/rawtx/test` to keep API design symmetric as we are dealing with rawtx not
   transactions
1. Moved `/transactions/estimate-fee` to `/fee/estimate`  to keep API design symmetric as we are dealing fee estimation
   not transactions

### CODE REVIEW

File: packages/whale-api-client/src/api/transactions.ts [hunk 1]
```diff
@@ -21,30 +14,26 @@ export class Transactions {
...
+    return await this.client.requestList('GET', `transactions/${txid}/vins`, size, next)
   }
 
   /**
-   * @param {number} confirmationTarget in blocks till fee get confirmed
-   * @return {Promise<number>} fee rate per KB
+   * @param {string} txid of the transaction
+   * @param {number} [size=30] size to query
+   * @param {string} [next] next token for next slice of vout
+   * @return {Promise<ApiPagedResponse<TransactionVin[]>>}
```
Comment (jingyi2811) [inline packages/whale-api-client/src/api/transactions.ts:30]: Return type does not match

File: src/module.api/rawtx.controller.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    const hex = await createSignedTxnHex(container, 10, 9.9999)
+    await controller.test({
+      hex: hex
+    })
+  })
+
+  it('should accept valid txn with given maxFeeRate', async () => {
+    const hex = await createSignedTxnHex(container, 10, 9.995)
+    await controller.test({
+      hex: hex,
```
Comment (jingyi2811) [inline src/module.api/rawtx.controller.e2e.ts:47]: There are a lot of `hex: hex `in this PR, maybe just refactor into `hex`

File: packages/whale-api-client/src/api/transactions.ts (hunk 1 above)
Comment (fuxingloh) [reply packages/whale-api-client/src/api/transactions.ts:30]:
```suggestion
   * @return {Promise<ApiPagedResponse<TransactionVout[]>>}
```

File: src/module.api/rawtx.controller.e2e.ts (hunk 2 above)
Comment (fuxingloh) [reply src/module.api/rawtx.controller.e2e.ts:47]: Will do it next time

**Review by aikchun — APPROVED** [review]
LGTM

**Review by jingyi2811 — APPROVED** [review]

**Review by ivan-zynesis — APPROVED** [review]

**Pull request merged**